subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/ecb.c

```diff
@@ -17,39 +17,194 @@
 #define LOG_MODULE_NAME bt_ctlr_rv32m1_ecb
 #include "common/log.h"
 #include "hal/debug.h"
+#include "fsl_cau3_ble.h"
 
 void ecb_encrypt_be(u8_t const *const key_be, u8_t const *const clear_text_be,
 		    u8_t * const cipher_text_be)
 {
+	u8_t keyAes[16] __aligned(4);
+	status_t status;
+
+	cau3_handle_t handle;
+
+	memcpy(&keyAes, key_be, sizeof(keyAes));
+
+	/* CAU3 driver supports 4 key slots. */
+	handle.keySlot = kCAU3_KeySlot1;
+
+	/* After encrypt/decrypt req is sent to CAU3, the Host CPU will
+	 * execute WFE() until CAU3 signals task done by setting the event.
+	 */
+	handle.taskDone = kCAU3_TaskDonePoll;
+
+	/* Loads the key into CAU3's DMEM and expands the AES key schedule */
+	status = CAU3_AES_SetKey(CAU3, &handle, keyAes, sizeof(keyAes));
+	if (status != kStatus_Success) {
+		LOG_ERR("CAUv3 AES key set failed %d", status);
+		return;
+	}
+
+	status = CAU3_AES_Encrypt(CAU3, &handle, clear_text_be, cipher_text_be);
+	if (status != kStatus_Success) {
+		LOG_ERR("CAUv3 AES encrypt failed %d", status);
+		return;
+	}
 }
 
 void ecb_encrypt(u8_t const *const key_le, u8_t const *const clear_text_le,
 		 u8_t * const cipher_text_le, u8_t * const cipher_text_be)
 {
+	u8_t keyAes[16] __aligned(4);
+	u8_t clear[16];
+	u8_t cipher[16];
+	status_t status;
+
+	cau3_handle_t handle;
+
+	/* The security function e of the cryptographic toolbox in CAU as in STD
+	 * The most significant octet of key corresponds to key[0], the most
+	 * significant octet of plaintextData corresponds to in[0] and the most
+	 * significant octet of encryptedData corresponds to out[0] using the
+	 * notation specified in FIPS-197
+	 * Thus, reverse the input parameters that are LSB to MSB format (le)
+	 */
+	mem_rcopy((u8_t *)&keyAes, key_le, sizeof(keyAes));
+	mem_rcopy((u8_t *)&clear, clear_text_le, sizeof(clear));
```

Comment:
minor: unnecessary typecasts

Reply:
Unlike memcpy/memcmp, mem_rcopy does not have (void *) as type of the first parameter.
void mem_rcopy(u8_t *dst….
Typecasting due to warnings.

Reply:
Ok, understood.

Reply:
BTW, we alse have sys_memcpy_swap from byteorder.h, Would it make sense to use that here instead?